compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/ClassBodyTest.kt

```diff
@@ -0,0 +1,130 @@
+
+class ClassBodyTest : AnnotationSpec() {
+
+  @Test
+  fun `validate class body scope properties are correctly generated`() {
+    assertThis(CompilerTest(
+      config = { listOf(addMetaPlugins(ClassBodyPlugin())) },
+      code = { classBody() },
+      assert = { quoteOutputMatches(
+        """
+        | class ClassBodyScopeTest {
+        |
+        |   private val x = "x"
+        |   private val y = "y"
+        |   private val z = "z"
+        |
+        |   companion object {
+        |     fun init() = ClassBodyScopeTest()
+        |   }
+        |
+        |   fun x() = x
+        |   fun y() = y
+        |   fun z() = z
+        |   fun test() = 0
+        | }
+        """.source
+      )}
+    ))
+  }
+  
+  @Test
+  fun `validate enum body scope properties are correctly generated`() {
+    assertThis(CompilerTest(
+      config = { listOf(addMetaPlugins(EnumBodyPlugin())) },
+      code = { enumBody() },
+      assert = { quoteOutputMatches(
+        """
+        | enum class EnumBodyScopeTest {
+        |   FOO, BAR;
+        |
+        |   fun foo() = 0
+        |   fun test() = 0
+        | }
+        """.source
+      )}
+    ))
+  }
+  
+  @Test
+  fun `validate object body scope properties are correctly generated`() {
+    assertThis(CompilerTest(
+      config = { listOf(addMetaPlugins(ClassBodyPlugin())) },
+      code = { objectBody() },
+      assert = { quoteOutputMatches(
+        """
+        | object ObjectBodyScopeTest {
+        |
+        |   private val x = "x"
+        |   private val y = "y"
+        |   private val z = "z"
+        |
+        |   fun x() = x
+        |   fun y() = y
+        |   fun z() = z
+        |   fun test() = 0
+        | }
+        """.source
+      )}
+    ))
+  }
+  
+  private fun classBody(): Code.Source {
+    return """
+           | //metadebug
+           |
+           | class ClassBodyScopeTest {
+           |
+           |   private val x = "x"
+           |   private val y = "y"
+           |   private val z = "z"
+           |
+           |   companion object {
+           |     fun init() = ClassBodyScopeTest()
+           |   }
+           |
+           |   fun x() = x
+           |   fun y() = y
+           |   fun z() = z
+           | }
+           """.source
+  }
+  
+  private fun enumBody(): Code.Source {
+    return """
+           | //metadebug
+           |
+           | enum class EnumBodyScopeTest {
+           |   FOO, BAR;
+           |
+           |   fun foo() = 0
+           | }
+           """.source
+  }
+  
+  private fun objectBody(): Code.Source {
+    return """
+           | //metadebug
+           |
+           | object ObjectBodyScopeTest {
+           |
+           |   private val x = "x"
+           |   private val y = "y"
+           |   private val z = "z"
+           |
+           |   fun x() = x
+           |   fun y() = y
+           |   fun z() = z
+           | }
+           """.source
+  }
```

Comment:
This looks good! We're going to want to add more testing for increased robustness i.e. moved order of companion objects, missing components of the test, etc.